### ISSUE TITLE
Question about the task_bs parameter

### ISSUE THREAD

**Issue by MengyuanChen21** [ehsansaleh/code_dcf#1]
Hello authors,

First of all, thanks for the great work!

I am now working on a related work and I have a small question about the task_bs parameter in Line 159 of main.py.

`task_bs = 10  # The number of tasks to stack to each other for parallel optimization`

My question is that, are we allowed to make a modification on this parameter without harming the fairness of comparison? Or in other words, if we change this parameter, are we still conducting a fair comparison? Since if we change it to 100 or even 1000, the whole process will be accelerated a lot.

**Comment by ehsansaleh** [comment]
Hi there,

Thanks for your note and finding our work helpful!

TLDR: Yes; if your GPU memory allows it, you should be able to raise `task_bs` without affecting the downstream analyses.

We just set `task_bs=10` so that none of our experiments would get an OOM (out of memory) error. Since our settings included both demanding and lite ablation studies and we only had access to 16GB Nvidia GPUs at the time, we had to go with a small `task_bs`. The [`code_firth`](https://github.com/ehsansaleh/code_firth/blob/3942a09814160e9db47bf9c194e3b71ab53b0396/utils/io_utils.py#L213) sister repository performed an optimization routine for choosing a reasonable task batch size, but in this repository, we didn't have the time to write such a routine for dynamically picking the best `task_bs`. As you suggested, small `task_bs` values can lead to poor GPU utilization and longer sequential running times.

That being said, you should be able to freely set any value for `task_bs` to increase the hardware performance without any statistical risks to the downstream results in principle. If you happen to witness otherwise, please let us know so that we look further into this.

I'll try to clarify this point further in the documentation and the code. Thanks again for pointing this out.

Sincerely, Ehsan

**Comment by MengyuanChen21** [comment]
I Get it. Thank you very much for your time and consideration!
